### PR TITLE
test: add runtime testing for kotlin

### PR DESCRIPTION
**Pull request by SumantxD**
as instructed in https://github.com/asyncapi/modelina/issues/1334 , i have added the runtime testing for kotlin

### CODE REVIEW

File: test/runtime/runtime-kotlin.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import * as fs from 'fs';
```
Comment (jonaslagoni) [inline test/runtime/runtime-kotlin.ts:1]: This whole file needs to follow https://github.com/asyncapi/modelina/blob/master/test/runtime/runtime-java.ts

Just for kotlin instead 🙂

File: test/runtime/runtime-kotlin/src/test/kotlin/MyClass.kt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+
+package com.mycompany.app
+
+class MyClass {
```
Comment (jonaslagoni) [inline test/runtime/runtime-kotlin/src/test/kotlin/MyClass.kt:4]: Make sure all generated src are ignored in git.

File: test/runtime/runtime-kotlin/src/test/kotlin/MyClassTest.kt [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import org.junit.Test
+import kotlin.test.assertEquals
+
+class MyClassTest {
+    @Test
+    fun testPrintMessage() {
+        val myClass = MyClass()
+        val name = "John"
+        val expectedMessage = "Hello, $name"
+        myClass.printMessage(name)
```
Comment (jonaslagoni) [inline test/runtime/runtime-kotlin/src/test/kotlin/MyClassTest.kt:12]: After you change the generation script, you need to change this test setup a bit.

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
I haven't worked much Kotlin, but please make sure you add a gitignore file under `test/runtime/runtime-kotlin` that is Kotlin project specific 🙂

**Comment by jonaslagoni** [comment]
Also the CI script fails to be setup 🙂

File: test/runtime/runtime-kotlin/.gitignore [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# Gradle-specific files
+build/
+
+# Generated Src files
+src/test/kotlin/Address.kt
```
Comment (jonaslagoni) [inline test/runtime/runtime-kotlin/.gitignore:5]: Are you sure it's this file you want to ignore? 🤔 https://github.com/asyncapi/modelina/blob/0175034f1d5d3e422265b0ef8a799dd2bc4214a6/test/runtime/runtime-kotlin/src/test/kotlin/AddressTest.kt

File: package.json [hunk 5]
```diff
@@ -52,6 +52,7 @@
     "@asyncapi/parserV1": "npm:@asyncapi/parser@^1.17.2",
     "@netlify/plugin-nextjs": "4.32.2",
     "@types/jest": "^27.0.2",
+    "@types/node": "^20.2.5",
```
Comment (jonaslagoni) [inline package.json:55]: This should not be necessary

File: jest.config.js [hunk 6]
```diff
@@ -28,5 +28,7 @@ module.exports = {
     '<rootDir>/examples/TEMPLATE',
     '<rootDir>/test/generators/template',
     '<rootDir>/test/processors/TemplateInputProcessor.spec.ts'
-  ]
+  ],
+  preset: 'ts-jest',
+  testEnvironment: 'node'
```
Comment (jonaslagoni) [inline jest.config.js:33]: Why is this necessary?

File: test/runtime/runtime-kotlin.ts [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  currentDirPath,
+  './runtime-kotlin/src/test/kotlin'
+);
+
+// Define the TypeScript code as a string template
+const typescriptCode = `
+import com.fasterxml.jackson.annotation.JsonCreator
+import com.fasterxml.jackson.annotation.JsonProperty
+
+data class Address @JsonCreator constructor(
```
Comment (jonaslagoni) [inline test/runtime/runtime-kotlin.ts:16]: I think you are misunderstanding me.

In this file, you need to use Modelina, to generate the models, NOT manually string templating anything 🙂

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
Is the .gradle files necessary to commit? 🤔 7

**Comment by SumantxD** [comment]
@jonaslagoni I have made all the requested changes !! :partying_face:

File: package.json [hunk 8]
```diff
@@ -46,7 +46,8 @@
     "alterschema": "^1.1.2",
     "change-case": "^4.1.2",
     "openapi-types": "9.3.0",
-    "typescript-json-schema": "^0.53.0"
+    "typescript-json-schema": "^0.53.0",
+    "winston": "^3.9.0"
```
Comment (jonaslagoni) [inline package.json:50]: Why are you adding a new dependency?

File: test/runtime/runtime-kotlin/.gitignore [hunk 9]
```diff
@@ -0,0 +1,5 @@
+# Gradle-specific files
```
Comment (jonaslagoni) [inline test/runtime/runtime-kotlin/.gitignore:1]: I would add the official git ignore for gradle, and of course keep the kotlin specific ignore of build, as well as the src files.

https://github.com/github/gitignore/blob/main/Gradle.gitignore

Remember to remove all the files it now ignores and commit it, otherwise the files will keep being in the PR.

File: test/runtime/runtime-kotlin/.gitignore [hunk 10]
```diff
@@ -0,0 +1,5 @@
...
+build/
+
+# Generated Src files
+src/main/kotlin/com/mycompany/app/generic
```
Comment (jonaslagoni) [inline test/runtime/runtime-kotlin/.gitignore:5]: I would just use `src/main`, unless you explicitly know that you will have src files that is not generated 🙂

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
Two main comments, otherwise we are nearly there 💪 

Btw, when you resolve one of the review comments, make sure you also resolve them in GitHub, that way its easy to see you addressed the concerns 🙂

File: test/runtime/runtime-kotlin/.gitignore (hunk 9 above)
Comment (SumantxD) [reply test/runtime/runtime-kotlin/.gitignore:1]: it's a logging library for node i have removed it as it is not required

File: package.json (hunk 8 above)
Comment (SumantxD) [reply package.json]: it's a logging library for node, i have removed it as it is not required.

File: test/runtime/runtime-kotlin/.gitignore (hunk 10 above)
Comment (SumantxD) [reply test/runtime/runtime-kotlin/.gitignore]: done !!

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
You need to manually remove all files within `.gradle` 🙂 

@SumantxD it's always a good idea to review your own PR before asking for a review (or re-review), I still do it my self. This is to ensure that all the changes you are proposing is actually what you want, and you didn't forget anything 🙂

File: package-lock.json [hunk 11]
```diff
@@ -20,12 +20,14 @@
         "alterschema": "^1.1.2",
         "change-case": "^4.1.2",
         "openapi-types": "9.3.0",
-        "typescript-json-schema": "^0.53.0"
+        "typescript-json-schema": "^0.53.0",
+        "winston": "^3.9.0"
```
[the remark this answers is not in the document]
Comment (SumantxD) [reply package-lock.json]: done !!

**Comment by SumantxD** [comment]
> You need to manually remove all files within `.gradle` slightly_smiling_face
Screenshot from 2023-06-06 21-39-13

i have removed it manually is it is reflected in the 5'th commit , plz correct me it i am wrong :smiley_cat:

**Review by jonaslagoni — APPROVED** [review]
Nice @SumantxD, LGTM 👍

**Comment by jonaslagoni** [comment]
@all-contributors please add @SumantxD for test, infra, code
